### PR TITLE
Adding CompareAndSet to store interface and make stack state transiti…

### PR DESCRIPTION
**Pull request by bquenin**
## Verification

```
$ sudo ./swarm start
$ make test-storage
$ make test-stack
```

### CODE REVIEW

File: api/rpc/stack/parse.go [hunk 1]
```diff
@@ -3,9 +3,12 @@ package stack
 import (
 	"strings"
 
-	"gopkg.in/yaml.v2"
-
+	"context"
```
Comment (generalhenry) [inline api/rpc/stack/parse.go]: shouldn't that be "golang.org/x/net/context"

**Comment by subfuzion** [comment]
Great work on the initial implementation. Will merge so it's in place for now for stack, and you can work on refactoring the implementation out of stack to create a generic state machine service that can be initialized with a state transition table and used by stack, service, etc.

👍

**Review by subfuzion — APPROVED** [review]
Great work on the initial implementation. Will merge so it's in place for now for stack, and you can work on refactoring the implementation out of stack to create a generic state machine service that can be initialized with a state transition table and used by stack, service, etc.

👍

File: api/rpc/stack/parse.go (hunk 1 above)
Comment (subfuzion) [reply api/rpc/stack/parse.go]: context is in core now: https://golang.org/pkg/context/

**Pull request merged**